packages/framework-core/src/services/graphql/graphql-query-generator.ts

```diff
@@ -145,6 +166,23 @@ export class GraphQLQueryGenerator {
     })
     return { filter: { type: filter } }
   }
+  public generateFilterListedQueriesFields(name: string, type: TargetTypeMetadata): GraphQLFieldConfigArgumentMap {
+    const filterArguments = this.generateFilterArguments(type)
+    const filter: GraphQLInputObjectType = new GraphQLInputObjectType({
+      name: `List${name}Filter`,
+      fields: () => ({
+        ...filterArguments,
+        and: { type: new GraphQLList(filter) },
+        or: { type: new GraphQLList(filter) },
+        not: { type: filter },
```

Comment:
if this type is `filter`, is it also `GraphQLInputObjectType`? Where does `filter` come from?

Reply:
It's weird that you are declaring the const filter and using it when you construct class instances inside the assignment

Reply:
`filter` is the actual const declared being used recursively to build the schema. It's weird but you can see how it's used [here](https://graphql.org/graphql-js/type/#graphqlobjecttype).

Reply:
I was surprised too, but it is how you can define recursive types in the GraphQL schema when using the programmatic API 🤯